Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/RideRequestUtils.java

```diff
@@ -264,4 +269,74 @@ public void openApplication(Context context){
         }catch (Exception ignored){}
     }
 
+    private static void updateStorage(String key, String value, Context context){
+        SharedPreferences sharedPref = context.getSharedPreferences(
+                context.getString(R.string.preference_file_key), Context.MODE_PRIVATE);
+        SharedPreferences.Editor editor = sharedPref.edit();
+        editor.putString(key, value);
+        editor.apply();
+    }
+
+    public void updateDriverStatus(Boolean status, String mode, Context context, Boolean startWidget) {
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        executor.execute(() ->
+        {
+            StringBuilder result = new StringBuilder();
+            SharedPreferences sharedPref = context.getSharedPreferences(context.getString(R.string.preference_file_key), Context.MODE_PRIVATE);
+            String token = sharedPref.getString("REGISTERATION_TOKEN", "null");
+            String bundle_version = sharedPref.getString("BUNDLE_VERSION","null");
+            String baseUrl = sharedPref.getString("BASE_URL", "null");
+            String deviceDetails = sharedPref.getString("DEVICE_DETAILS", "null");
+            try
+            {
+                //endPoint for driver status
+                String orderUrl = baseUrl + "/driver/setActivity?active=" + status + "&mode=\"" + mode + "\"";
+                Log.d(LOG_TAG, "orderUrl " + orderUrl);
+                //Http connection to make API call
+                HttpURLConnection connection = (HttpURLConnection) (new URL(orderUrl).openConnection());
+                if (connection instanceof HttpsURLConnection)
+                    ((HttpsURLConnection) connection).setSSLSocketFactory(new TLSSocketFactory());
+                connection.setRequestMethod("POST");
+                connection.setRequestProperty("Content-Type", "application/json");
+                connection.setRequestProperty("x-client-version", versionName);
+                connection.setRequestProperty("token", token);
+                connection.setRequestProperty("x-bundle-version", bundle_version);
+                connection.setRequestProperty("x-device", deviceDetails);
+                connection.setDoOutput(true);
+                connection.connect();
+
+                // validating the response code
+                int respCode = connection.getResponseCode();
+                InputStreamReader respReader;
+                Log.d(LOG_TAG, "respCode "+ respCode);
+
+                if(respCode == 200 && startWidget == true){
```

Comment:
no need to create a new if else case it's already there
Just move this to line number 324